Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -395,16 +401,16 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
 
         let pbxBuildFile = PBXBuildFile(file: modelReference)
         pbxproj.add(object: pbxBuildFile)
-        buildPhase.files?.append(pbxBuildFile)
+        return pbxBuildFile
```

Comment:
minor: seeing the method now returns build files, perhaps we can shift the responsibility of `pbxproj.add(object: pbxBuildFile)` to the caller? This will help keep the integration with the `pbxproj` / build phase local to one place. 

Also helps the `map` operations that leverage this method remain side effect free (i.e. can safely be called multiple times)

(same for the other methods here where applicable)